cloud/vsphere/machineactuator.go

```diff
@@ -506,69 +497,123 @@ func (vc *VsphereClient) Update(cluster *clusterv1.Cluster, goalMachine *cluster
 	}
 
 	if util.IsMaster(goalMachine) {
+		// Master upgrades
 		glog.Info("Upgrade for master machine.. Check if upgrade needed.")
-		// Get the versions for the new object.
-		goalMachineControlPlaneVersion := goalMachine.Spec.Versions.ControlPlane
-		goalMachineKubeletVersion := goalMachine.Spec.Versions.Kubelet
 
 		// If the saved versions and new versions differ, do in-place upgrade.
-		if goalMachineControlPlaneVersion != goalMachine.Annotations[ControlPlaneVersionAnnotationKey] ||
-			goalMachineKubeletVersion != goalMachine.Annotations[KubeletVersionAnnotationKey] {
-			glog.Infof("Doing in-place upgrade for master from %s to %s", goalMachine.Annotations[ControlPlaneVersionAnnotationKey], goalMachineControlPlaneVersion)
+		if vc.needsMasterUpdate(goalMachine) {
+			glog.Infof("Doing in-place upgrade for master from v%s to v%s", goalMachine.Annotations[ControlPlaneVersionAnnotationKey], goalMachine.Spec.Versions.ControlPlane)
 			err := vc.updateMasterInPlace(goalMachine)
 			if err != nil {
-				glog.Errorf("Master inplace update failed: %+v", err)
+				glog.Errorf("Master in-place upgrade failed: %+v", err)
 			}
 		} else {
-			glog.Info("UPDATE TYPE NOT SUPPORTED")
+			glog.Info("UNSUPPORTED MASTER UPDATE.")
 			return nil
```

Comment:
why does this branch return nil? Others don't and they all fall out to the nil return at the bottom.

Reply:
Actually, it seems that there isn't a return value other than nil. Why does the func have a return type at all?